.github/workflows/action-deploy-apps.yml

```diff
@@ -0,0 +1,175 @@
+﻿name: Deploy apps
+
+env:
+  AZ_CLI_VERSION: 2.56.0
+
+on:
+  workflow_call:
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      AZURE_SUBSCRIPTION_ID:
+        required: true
+      AZURE_RESOURCE_GROUP_NAME:
+        required: true
+      AZURE_ENVIRONMENT_KEY_VAULT_NAME:
+        required: true
+      AZURE_CONTAINER_APP_ENVIRONMENT_ID:
+        required: true
+      AZURE_APP_INSIGHTS_CONNECTION_STRING:
+        required: true
+      AZURE_APP_CONFIGURATION_NAME:
+        required: true
+      AZURE_ADO_CONNECTION_STRING_SECRET_URI:
+        required: true
+
+    inputs:
+      region:
+        required: true
+        type: string
+      environment:
+        required: true
+        type: string
+      dryRun:
+        description: "If true, the deployment will be validated but not executed."
+        required: false
+        type: boolean
+        default: false
+      gitShortSha:
+        description: "Short SHA of the commit"
+        required: true
+        type: string
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.environment }}
+  # if the dryrun input is true, we want to cancel any running deployments in order to not block the pipeline
+  cancel-in-progress: ${{ inputs.dryRun }}
```

Comment:
Hm, `action-deploy-apps.yml`, hvis det kjører andre deploys av denne actionene mot samme environment? 
Blir dette å stoppe ikke-dryRun-deploys mot dette env.?

Reply:
Slik jeg forstår det:
`group: ${{ github.workflow }}-${{ inputs.environment }}` sørger for at kun en av deploysene per miljø kjører samtidig, så vi ikke får flere samtidige forsøk på deploys. 

`cancel-in-progress: ${{ inputs.dryRun }}`

Det var egentlig lagt til for å cancelle jobber som blir stående i review om det kommer en ny commit. F.eks om vi ruller ut til `main`, deployer til `test` og `dryrun` kjøres for `staging` med approval. Om vi får inn en ny commit da, så må vi ned og approve den forrige som avventet approval for `staging`. Om det ga mening 🤔

Mulig det er noen fallgruver her jeg ikke har tenkt helt ut. Men ser for meg at dette løser det use-caset iallfall (testet forsåvidt tidligere også)

Reply:
Mhm
Men så for meg at dette og kan skje:

PR 1 blir merget til main, starter deploy mot test (ikke dryRun)
PR 2 blir opprettet akkurat da, og starter en dryRun mot test

Blir deployen til PR 1 stoppet da?

Reply:
Vel egentlig ikke fordi vi ikke kjører dryrun mot test i `main`. Men ser et annet issue her ja. At PR dryruns kan cancelle hverandre :D Så burde nok ha med branch-navn her også!

Reply:
Tok en prat på DM. Endte opp med følgende:

```
# Existing runs are cancelled if someone repeatedly commits to their own Pull Request (PR). However, it does not stop others' dry runs or actual deployments from the main branch.
  # Also, the cancellation does not occur on merges to the main branch. Therefore, if multiple merges to main are performed simultaneously, they will just be queued up.
  group: ${{ github.workflow }}-${{ inputs.environment }}-${{ github.ref_name }}-${{ inputs.dryRun }}
  # if the dryrun input is true, we want to cancel any running deployments in order to not block the pipeline e.g for environment approvals
  cancel-in-progress: ${{ inputs.dryRun }}
```